test/libsolidity/smtCheckerTests/operators/ternary_operator_3.sol

```diff
@@ -0,0 +1,31 @@
+abstract contract D {
+	function d() external virtual returns (uint);
+}
+
+contract C {
+
+  D d;
+  uint v;
+  bool guard = true;
+
+  function inc() public {
+	++v;
+  }
+
+  function dec() public {
+    if (guard) return;
+    --v;
+  }
+
+  function f() public returns (uint) {
+    guard = false;
+    uint ret = v > 0 ? d.d() : 0;
+	guard = true;
+	return ret;
+  }
+}
+// ====
+// SMTEngine: chc
+// SMTTargets: underflow
+// ----
+// Warning 3944: (215-218): CHC: Underflow (resulting value less than 0) happens here.
```

Comment:
I guess the counterexample here is `d.d()` calling `dec()` twice or more?

Reply:
Yes, first `inc()` has to be called at least once, then `d.d()` can call `dec` one additional time and underflow will be triggered. Can we enable output of a full counterexample?